Gems/Atom/Tools/AtomToolsFramework/Code/Source/AssetBrowser/AtomToolsAssetBrowserInteractions.cpp

```diff
@@ -83,6 +84,18 @@ namespace AtomToolsFramework
     void AtomToolsAssetBrowserInteractions::AddContextMenuActionsForSourceEntries(
         [[maybe_unused]] QWidget* caller, QMenu* menu, const AzToolsFramework::AssetBrowser::AssetBrowserEntry* entry)
     {
+        QFileInfo fileInfo(entry->GetFullPath().c_str());
+        QString extension = fileInfo.completeSuffix();
+        if (extension == "shader")
+        {
+            menu->addAction("Create new variantlist", [entry]()
```

Comment:
Same here, I'd need a "dependency inversion" method where the top tool can "inject" an action in this menu.

Reply:
What I'd find most meaningful in terms of function and UX is that the .shader is NOT supported. This is not a valid source document type. The document type is meant to be .shadervariantlist and that's it.
BUT, `.shader` is indeed the side-file type to extract the option list and create an empty .shadervariantlist starter document.
I'd ideally mean to make this distinction clear so that .shader opening just end up in the json opening in a text editor.
But right click "generate new doc from..." (currently phrased "create new variantlist") would work.
Maybe I need to create a phony .shadervariantlist placeholder with just a "please kickstart me from <path.shader>" flag data inside only, and then the SMC::Open can "parse" that.